subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginCheckInIntegrationTest.groovy

```diff
@@ -121,8 +124,36 @@ class GradleEnterprisePluginCheckInIntegrationTest extends AbstractIntegrationSp
         getMinimumPluginVersionForConfigurationCaching() | true
     }
 
+    @Requires(IntegTestPreconditions.NotConfigCached)
+    def "shows warning message when Gradle Enterprise plugin version is used with isolated projects enabled"() {
+        given:
+        plugin.runtimeVersion = plugin.artifactVersion = pluginVersion
+        applyPlugin()
+        settingsFile << """
+            println "present: " + services.get($GradleEnterprisePluginManager.name).present
+        """
+
+        when:
+        succeeds("t", "-D${StartParameterBuildOptions.IsolatedProjectsOption.PROPERTY_NAME}=true")
+
+        then:
+        output.contains("present: ${applied}")
+
+        and:
+        output.contains("gradleEnterprisePlugin.checkIn.unsupported.reasonMessage = $UNSUPPORTED_PLUGIN_DUE_TO_ISOLATED_PROJECTS_MESSAGE") != applied
+
+        where:
+        pluginVersion                                | applied
+        '3.11.4'                                     | false
```

Comment:
🤔 I would suggest to also add a smoke test in `BuildScanPluginSmokeTest` for the case when PI is enabled. This will allow us to better see the impact of this and future changes in the cross-version context.

Reply:
@alllex I've added a test, but can you please verify that it is what you expected?

Reply:
I ended up polishing the smoke test myself to move this PR faster, so we can merge it before 8.3. See the comment on the PR itself.